langserver/completion.go

```diff
@@ -155,41 +158,50 @@ func (s *server) completeFunctionName(completions *[]protocol.CompletionItem, lo
 		return err
 	}
 
+	names := make([]string, len(promql.Functions))
```

Comment:
you could compact a bit these three lines like that:

```go
var names []string
for name := range promql.Functions {
    names = append(names, strings.ToLower(name))
}
```

Reply:
this logic can be applied in the whole file ;)

Reply:
names is being reused at most once in a function ,so it might become necessary to clear names to add new data which we can avoid. also append may allocate more byte than needed.

Reply:
well it is still valid, you can still clear the variable names with the logic described above.

And beside, you could also factorize this logic of extraction too.

Reply:
You could could use pre-allocation, to already reserve the right amount of memory:

```go
names := make([]string, 0, len(promql.Functions))
for name := range promql.Functions {
    names = append(names, strings.ToLower(name))
}
```

Reply:
(Not that it really matters. The map is way more expensive anyway.)